command/agent/config.go

```diff
@@ -26,344 +26,362 @@ import (
-	Enabled bool `mapstructure:"enabled"`
+	Enabled bool `hcl:"enabled"`
 
 	// StateDir is the state directory
-	StateDir string `mapstructure:"state_dir"`
+	StateDir string `hcl:"state_dir"`
 
 	// AllocDir is the directory for storing allocation data
-	AllocDir string `mapstructure:"alloc_dir"`
+	AllocDir string `hcl:"alloc_dir"`
 
 	// Servers is a list of known server addresses. These are as "host:port"
-	Servers []string `mapstructure:"servers"`
+	Servers []string `hcl:"servers"`
 
 	// NodeClass is used to group the node by class
-	NodeClass string `mapstructure:"node_class"`
+	NodeClass string `hcl:"node_class"`
 
 	// Options is used for configuration of nomad internals,
 	// like fingerprinters and drivers. The format is:
 	//
 	//  namespace.option = value
-	Options map[string]string `mapstructure:"options"`
+	Options map[string]string `hcl:"options"`
 
 	// Metadata associated with the node
-	Meta map[string]string `mapstructure:"meta"`
+	Meta map[string]string `hcl:"meta"`
 
 	// A mapping of directories on the host OS to attempt to embed inside each
 	// task's chroot.
-	ChrootEnv map[string]string `mapstructure:"chroot_env"`
+	ChrootEnv map[string]string `hcl:"chroot_env"`
 
 	// Interface to use for network fingerprinting
-	NetworkInterface string `mapstructure:"network_interface"`
+	NetworkInterface string `hcl:"network_interface"`
 
 	// NetworkSpeed is used to override any detected or default network link
 	// speed.
-	NetworkSpeed int `mapstructure:"network_speed"`
+	NetworkSpeed int `hcl:"network_speed"`
 
 	// CpuCompute is used to override any detected or default total CPU compute.
-	CpuCompute int `mapstructure:"cpu_total_compute"`
+	CpuCompute int `hcl:"cpu_total_compute"`
 
 	// MemoryMB is used to override any detected or default total memory.
-	MemoryMB int `mapstructure:"memory_total_mb"`
+	MemoryMB int `hcl:"memory_total_mb"`
 
 	// MaxKillTimeout allows capping the user-specifiable KillTimeout.
-	MaxKillTimeout string `mapstructure:"max_kill_timeout"`
+	MaxKillTimeout string `hcl:"max_kill_timeout"`
 
 	// ClientMaxPort is the upper range of the ports that the client uses for
 	// communicating with plugin subsystems
-	ClientMaxPort int `mapstructure:"client_max_port"`
+	ClientMaxPort int `hcl:"client_max_port"`
 
 	// ClientMinPort is the lower range of the ports that the client uses for
 	// communicating with plugin subsystems
-	ClientMinPort int `mapstructure:"client_min_port"`
+	ClientMinPort int `hcl:"client_min_port"`
 
 	// Reserved is used to reserve resources from being used by Nomad. This can
 	// be used to target a certain utilization or to prevent Nomad from using a
 	// particular set of ports.
-	Reserved *Resources `mapstructure:"reserved"`
+	Reserved *Resources `hcl:"reserved"`
 
 	// GCInterval is the time interval at which the client triggers garbage
 	// collection
-	GCInterval time.Duration `mapstructure:"gc_interval"`
+	GCInterval    time.Duration
+	GCIntervalHCL string `hcl:"gc_interval"`
```

Comment:
I would use json tags here as well to keep `GCInterval` refer to string value in our API and omit the other one.

Reply:
I'm not sure I follow here, the time.Duration value looks like it's the one used by the api in master

Reply:
Ignore me - I got confused here - indeed you are preserving the json behavior here and rename only affects hcl parsing.